forge-gui/res/cardsfolder/upcoming/scion_of_halaster.txt

```diff
@@ -0,0 +1,15 @@
+Name:Scion of Halaster
+ManaCost:1 B
+Types:Legendary Enchantment Background
+S:Mode$ Continuous | Affected$ Creature.IsCommander+YouOwn | AddReplacementEffects$ Draw | Description$ Commander creatures you own have "The first time you would draw a card each turn, instead look at the top two cards of your library. Put one of them into your graveyard and the other back on top of your library. Then draw a card."
+SVar:Draw:Event$ Draw | ReplaceWith$ DBDig | CheckSVar$ X | SVarCompare$ EQ0 | CheckSecondSVar$ Y | SecondSVarCompare$ EQ0 | Description$ The first time you would draw a card each turn, instead look at the top two cards of your library. Put one of them into your graveyard and the other back on top of your library. Then draw a card.
+SVar:DBDig:DB$ Dig | DigNum$ 2 | AnyNumber$ | DestinationZone$ Graveyard | LibraryPosition2$ 0 | SubAbility$ DBStoreNum
+SVar:TrigReset:DB$ StoreSVar | SVar$ Y | Type$ Number | Expression$ 1
```

Comment:
I'm not sure that's good enough

What if the commander changes controller mid turn?
The Y will prevent draw replacement for the other player even if he didn't draw any yet?

Reply:
Good call – I think I have a better solution